pkg/sdk/symbols/info/info.go

```diff
@@ -105,7 +107,27 @@ func plugin_get_required_api_version() *C.char {
 	return (*C.char)(pRequiredAPIVersion.CharPtr())
 }
 
+func string_split(version string) (string, string, string) {
+	nums := strings.Split(version, ".")
+	if len(nums) != 3 {
+		panic("Incorrect format. Expected: Semantic Versioning: X.Y.Z")
+	}
+	return nums[0], nums[1], nums[2]
+}
+
 func SetRequiredAPIVersion(apiVer string) {
+	requiredMajor, requiredMinor, requiredPatch := string_split(apiVer)
+	pluginMajor, pluginMinor, pluginPatch := string_split(C.GoString(plugin_get_required_api_version()))
+
+	if pluginMajor != requiredMajor {
+		panic("Plugin sdk's Major version disagrees. Expected: Major version should be equal to " + pluginMajor + "but got " + requiredMajor)
```

Comment:
`Plugin SDK Go required API version incompatible major number.`
I think you could reuse this prefix below as well.